File: dashboard/src/lib/card.ts
export class Card {
    card_number: string;
    card_type: string;
    exp_month: number;
    exp_year: number;
    cvv: string;

    constructor({ card_number, card_type, exp_month, exp_year, cvv }: { card_number: string; card_type: string; exp_month: number; exp_year: number; cvv: string }) {
        this.card_number = card_number;
        this.card_type = card_type;
        this.exp_month = exp_month;
        this.exp_year = exp_year;
        this.cvv = cvv;
    }
}

File: frauddetection/src/main/java/spendreport/dtos/Card.java
import com.fasterxml.jackson.annotation.JsonProperty;

@Data
public class Card {
    @JsonProperty("card_number")
    private String cardNumber;
    @JsonProperty("card_type")
    private String cardType;
    @JsonProperty("exp_month")
    private int expMonth;
    @JsonProperty("exp_year")
    private int expYear;
    @JsonProperty("cvv")
    private String cvv;
}


